Helper/Api/Products.php

```diff
@@ -430,38 +430,33 @@ public function updateProductLinks(
     public function exists($storeId, array $result)
     {
         $storekeeper_id = $this->getResultStoreKeeperId($result);
-        try {
-            $collection = $this->productCollectionFactory->create();
-            $collection
-                ->addAttributeToSelect('*')
-                ->addAttributeToFilter('storekeeper_product_id', $storekeeper_id)
-                ->setFlag('has_stock_status_filter', false);
-
-            if (is_array($result) && isset($result['flat_product']) && isset($result['flat_product']['product'])) {
-                $collection->addAttributeToFilter(
-                    'sku', $result['flat_product']['product']['sku'] ?? null
-                );
-            }
+        $collection = $this->productCollectionFactory->create();
+        $collection
+            ->addAttributeToSelect('*')
+            ->addAttributeToFilter('storekeeper_product_id', $storekeeper_id)
+            ->setFlag('has_stock_status_filter', false);
+
+        if (is_array($result) && isset($result['flat_product']) && isset($result['flat_product']['product'])) {
+            $collection->addAttributeToFilter(
+                'sku', $result['flat_product']['product']['sku'] ?? null
+            );
+        }
 
-            if ($collection->count()) {
-                $firstItem = $collection->getFirstItem();
-                return $firstItem;
-            }
+        if ($collection->count()) {
+            $firstItem = $collection->getFirstItem();
+            return $firstItem;
+        }
 
-            try {
-                $storekeeper_sku = $this->getResultSku($result);
+        try {
+            $storekeeper_sku = $this->getResultSku($result);
 
-                if ($result = $this->productRepository->get($storekeeper_sku)) {
-                    return $result;
-                }
-            } catch (\Exception $e) {
-                // ignoring
+            if (!is_null($storekeeper_sku)) {
+                return $this->productRepository->get($storekeeper_sku);
             }
-
-            return false;
         } catch (\Exception $e) {
-            $this->logger->error($e->getMessage(), $this->logger->buildReportData($e));
+            // ignoring
```

Comment:
why ignoring? we should never ignore exceptions expecially when catching `\Exception`